### PR TITLE
feat: add create adhoc query component

### PR DESCRIPTION
**Pull request by OussEmaDevCode**
## Description 
Added the create adhoc query component, connected it via routing, designed its interface and connected it to the API.

## Related issues and discussion
#537 

## Screenshots, if any
Capture d’écran (238)

## Checklist
Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] If you have multiple commits please combine them into one commit by squashing them.

- [x] Read and understood the contribution guidelines at `web-app/.github/CONTRIBUTING.md`.

### CODE REVIEW

**Comment by OussEmaDevCode** [comment]
@abhaychawla Hi, please have a look.

File: src/app/organization/adhoc-query/create-adhoc-query/create-adhoc-query.component.html [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+          </mat-form-field>
+
+          <mat-form-field *ngIf="adhocQueryForm.controls.reportRunFrequency.value === 5">
+            <mat-label>Number of Days to run report</mat-label>
+            <input type="number" matInput required formControlName="reportRunEvery">
+            <mat-error *ngIf="adhocQueryForm.controls.reportRunEvery.hasError('required')">
+              Number of Days to run report is <strong>required</strong>
+            </mat-error>
+            <mat-error *ngIf="adhocQueryForm.controls.reportRunEvery.hasError('min')">
+              Number of Days to run report <strong>should</strong> be larger than 0
```
Comment (punwai) [inline src/app/organization/adhoc-query/create-adhoc-query/create-adhoc-query.component.html]:
```suggestion
              Number of Days to run report <strong>must be larger than 0</strong>
```

Comment (OussEmaDevCode) [reply src/app/organization/adhoc-query/create-adhoc-query/create-adhoc-query.component.html]: I'll wait for the mentors review and if it's a good suggestion, I'll incorporate it in the next push.

Comment (punwai) [reply src/app/organization/adhoc-query/create-adhoc-query/create-adhoc-query.component.html]: I'm just suggesting this according to my previous PRs, Abhay suggested this to me on that PR (I can't remember which one sorry)

File: src/app/organization/adhoc-query/create-adhoc-query/create-adhoc-query.component.html [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+            <mat-error *ngIf="adhocQueryForm.controls.query.hasError('required')">
+              SQL Query is <strong>required</strong>
+            </mat-error>
+          </mat-form-field>
+
+          <mat-form-field>
+            <mat-label>Insert Into table</mat-label>
+            <input matInput required formControlName="tableName">
+            <mat-error *ngIf="adhocQueryForm.controls.tableName.hasError('required')">
+              Insert Into table is <strong>required</strong>
```
Comment (abhaychawla) [inline src/app/organization/adhoc-query/create-adhoc-query/create-adhoc-query.component.html]:
```suggestion
              Insert into table is <strong>required</strong>
```

File: src/app/organization/adhoc-query/create-adhoc-query/create-adhoc-query.component.html [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+            <input matInput formControlName="email">
+            <mat-error *ngIf="adhocQueryForm.controls.email.hasError('email')">
+              Email is <strong>invalid</strong>
+            </mat-error>
+          </mat-form-field>
+
+          <mat-form-field>
+            <mat-label>Report Run Frequency</mat-label>
+            <mat-select formControlName="reportRunFrequency">
+              <mat-option *ngFor="let reportRunFrequency of reportRunFrequenciesData" [value]="reportRunFrequency.id">
```
Comment (abhaychawla) [inline src/app/organization/adhoc-query/create-adhoc-query/create-adhoc-query.component.html]:
```suggestion
              <mat-option *ngFor="let reportRunFrequency of reportRunFrequencyData" [value]="reportRunFrequency.id">
```

File: src/app/organization/adhoc-query/create-adhoc-query/create-adhoc-query.component.html [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+                {{ reportRunFrequency.value }}
+              </mat-option>
+            </mat-select>
+          </mat-form-field>
+
+          <mat-form-field *ngIf="adhocQueryForm.controls.reportRunFrequency.value === 5">
+            <mat-label>Number of Days to run report</mat-label>
```
Comment (abhaychawla) [inline src/app/organization/adhoc-query/create-adhoc-query/create-adhoc-query.component.html]:
```suggestion
            <mat-label>Custom Report Run Frequency (Days)</mat-label>
```

File: src/app/organization/adhoc-query/create-adhoc-query/create-adhoc-query.component.html (hunk 1 above)
Comment (abhaychawla) [reply src/app/organization/adhoc-query/create-adhoc-query/create-adhoc-query.component.html]: Update the message. Also, update the field label as I specified above.

File: src/app/organization/adhoc-query/create-adhoc-query/create-adhoc-query.component.html [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+            <mat-error *ngIf="adhocQueryForm.controls.reportRunEvery.hasError('required')">
+              Number of Days to run report is <strong>required</strong>
+            </mat-error>
+            <mat-error *ngIf="adhocQueryForm.controls.reportRunEvery.hasError('min')">
+              Number of Days to run report <strong>should</strong> be larger than 0
+            </mat-error>
+          </mat-form-field>
+
+          <mat-checkbox labelPosition="before" formControlName="isActive">
+            Active Status
```
Comment (abhaychawla) [inline src/app/organization/adhoc-query/create-adhoc-query/create-adhoc-query.component.html]:
```suggestion
            Active
```

**Comment by punwai** [comment]
@abhaychawla Considering that this is quite a large form (7 fields with 2 textareas) wouldn't it be better if it were to be double columned? Please don't implement it yet, but that's just what I did when I implemented this earlier on in the competition (I didn't get to claim the task nor push the PR in at the end) 
@OussEmaDevCode This can be seen implemented like in accounting rules

**Comment by abhaychawla** [comment]
@punwai Yes, let me think about it. Actually there are two text areas here so I am not sure.

**Comment by OussEmaDevCode** [comment]
@abhaychawla I'll wait for the response !

**Comment by abhaychawla** [comment]
We will merge this for now.

**Pull request merged**